plugins/module_utils/network/sonic/facts/qos_interfaces/qos_interfaces.py

```diff
@@ -85,6 +85,10 @@ def update_qos_interfaces(self, module):
                     scheduler_policy = intf['output']['scheduler-policy']['config']['name']
                     config_dict['scheduler_policy'] = scheduler_policy
 
+                if ('openconfig-qos-buffer:cable-length' in intf and 'config' in intf['openconfig-qos-buffer:cable-length']):
+                    cable_length = intf['openconfig-qos-buffer:cable-length']['config']['length']
```

Comment:
Please add protection and appropriate handling for the case in which the ...['length'] does not exist. (Generating an exception should be avoided for that case and, if it occurs, no cable length config entry should be added.)

This could be done by making this instruction a "get" and checking the result of checking to see if 'length' is "in" intf['openconfig-qos-buffer:cable-length']['config'].